### PR TITLE
fix(es/minifier): Don't drop inlined parameter as duplicate

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

Previously, we prepend a variable declaration using IIFE invoker, and dropped it as a duplicate because it's declared as a function parameter.

**Related issue:**

 - https://github.com/vercel/next.js/discussions/30237#discussioncomment-3983699

---

# Investigation

````
  DEBUG  Removing a variable statement because it's a function parameter, kind: "change"
    at crates/swc_ecma_minifier/src/compress/optimize/unused.rs:765
    in visit_mut_stmt
    in handle_stmt_likes
    in visit_mut_function
    in visit_mut_prop
    in seq_expr_with_children
    in visit_mut_seq_expr
    in visit_mut_call_expr
    in visit_mut_arrow_expr
    in seq_expr_with_children
    in visit_mut_seq_expr
    in visit_mut_expr_stmt
    in visit_mut_stmt
    in handle_stmt_likes
    in apply full optimizer
    in optimize with pass: 1
    in compress ast
    in minify

```

### CODE REVIEW

File: crates/swc_ecma_minifier/tests/fixture/next/tinymce/1/input.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Yp.revoke, (e => Yp.config((e => ({
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/fixture/next/tinymce/1/input.js:1]: Previous output of this file:

```js
Yp.revoke, (e1)=>Yp.config({
        aria: {
            mode: "checked"
        },
        ...ge(e1, (e1, t1)=>"exclusive" !== t1),
        onToggled (t1, o) {
            p(e1.onToggled) && e1.onToggled(t1, o), e = t1, t = o, Or(e, th, {
                item: e,
                state: t
            });
        }
    });

```

File: crates/swc_ecma_minifier/src/compress/optimize/unused.rs [hunk 2]
```diff
@@ -759,9 +759,14 @@ where
         for d in var.decls.iter_mut() {
             if d.init.is_none() {
                 if let Pat::Ident(name) = &d.name {
-                    if let Some(usage) = self.data.vars.get(&name.to_id()) {
-                        if usage.is_fn_local && usage.declared_as_fn_param {
+                    if let Some(usage) = self.data.vars.get_mut(&name.to_id()) {
+                        if usage.is_fn_local
+                            && usage.declared_as_fn_param
+                            && usage.declared_count >= 2
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/compress/optimize/unused.rs:765]: This is the fix

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/tests/fixture/next/tinymce/1/output.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+            mode: "checked"
+        },
+        ...ge(a, (e, t)=>"exclusive" !== t),
+        onToggled (c, d) {
+            p(a.onToggled) && a.onToggled(c, d), Or(c, th, {
+                item: c,
+                state: d
+            });
+        }
+    });
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/fixture/next/tinymce/1/output.js:12]: Previous output:

```js
"use strict";
Yp.revoke, (o)=>Yp.config({
        aria: {
            mode: "checked"
        },
        ...ge(o, (o, g)=>"exclusive" !== g),
        onToggled (g, t) {
            p(o.onToggled) && o.onToggled(g, t), e = g, f = t, Or(e, th, {
                item: e,
                state: f
            });
        }
    });

```

File: crates/swc_ecma_codegen/benches/bench.rs [hunk 4]
```diff
@@ -114,7 +114,7 @@ fn bench_emitter(b: &mut Bencher, s: &str) {
                 let _ = emitter.emit_module(&module);
             }
             black_box(buf);
-            let srcmap = cm.build_source_map(&mut src_map_buf);
+            let srcmap = cm.build_source_map(&src_map_buf);
```
Comment (jridgewell) [inline crates/swc_ecma_codegen/benches/bench.rs:117]: 😅

**Review by jridgewell — APPROVED** [review]

**Pull request merged**